system/modules/core/widgets/FileSelector.php

```diff
@@ -456,6 +456,12 @@ protected function convertValuesToPaths()
 			return;
 		}
 
+		// Do not convert values if path is not in the upload path
+		if ($this->path !== \Config::get('uploadPath'))
```

Comment:
Ah, I see. Still, what if `$this->path` is a subfolder of `Config::get('uploadPath')`? We must only return if the path is outside the upload directory IMHO.